src/mongo/tree/MongoDocumentTreeItem.ts

```diff
@@ -64,7 +64,10 @@ export class MongoDocumentTreeItem implements IAzureTreeItem {
 
     public static async update(collection: Collection, newDocument: IMongoDocument): Promise<IMongoDocument> {
         const filter: object = { _id: new ObjectID(newDocument._id) };
-        await collection.updateOne(filter, _.omit(newDocument, '_id'));
+        const result = await collection.updateOne(filter, _.omit(newDocument, '_id'));
+        if (result.upsertedCount != 1) {
+            throw new Error("Document not found. Please check if ID exists.");
```

Comment:
We don't know if "Document not found" is the reason it failed. It could have been any number of reasons.

What if we generalize it a little bit: ``Failed to update document with _id "${newDocument._id}".``